torchao/prototype/intx/Intx.py

```diff
@@ -0,0 +1,208 @@
+        return f"Int{self.bit_size}Tensor(shape = {self.packed_shape}, data = {unpack(self.get_shards(), self.bit_size, dim = self.pack_dim)})"
+    
+    def __tensor_flatten__(self):
+        
+        return self.__class__.bits_to_shard[self.bit_size], [self.packed_shape, self.bit_size, self.pack_dim]
+    @classmethod
+    def __tensor_unflatten__(
+        cls, tensor_data_dict, tensor_attributes, outer_size, outer_stride
+    ):
+        shards = [i for i in tensor_data_dict.values()]
+        packed_shape, bit_size, pack_dim = tensor_attributes
+        return cls(shards, packed_shape, bit_size, pack_dim)
+
+    @classmethod
+    def __torch_function__(cls, func, types, args=(), kwargs=None):
+        kwargs = {} if kwargs is None else kwargs
+
+        if func in _ATEN_OP_OR_TORCH_FN_TABLE[cls]:
+            return _ATEN_OP_OR_TORCH_FN_TABLE[cls][func](*args, **kwargs)
+
+        with torch._C.DisableTorchFunctionSubclass():
+            return func(*args, **kwargs)
+
+    @classmethod
+    def __torch_dispatch__(cls, func, types, args, kwargs):
+
+        if func in _ATEN_OP_OR_TORCH_FN_TABLE[cls]:
+            return _ATEN_OP_OR_TORCH_FN_TABLE[cls][func](func, *args, **kwargs)
+
+        raise NotImplementedError(
+            f"IntxTensor dispatch: attempting to run {func}, this is not supported"
+        )
+
+    __torch_function__ = torch._C._disabled_torch_function_impl
+
+    def get_plain(self):
+        return unpack(self.get_shards(), self.bit_size, dim = self.pack_dim)
+
+    # temporary until kernels on packed tensors are created
+    def apply_transformation(self, fn):
+        og = self.get_plain()
+        new = fn(og)
+        return self.from_int(new, self.bit_size, self.pack_dim)
+    
+    @classmethod
+    def from_int(cls, int_data: torch.Tensor, bit_size, pack_dim: int = -1):
+        shards = pack(int_data, bit_size, dim=pack_dim)
+        shape = list(int_data.shape)
+        shape[pack_dim] = shape[pack_dim] * bit_size // 8
+        return cls(shards, int_data.shape, bit_size, pack_dim)
+    
+    
+@implements([aten.to.device])
+def to(self, *args, **kwargs):
+    return self.__class__(
+        [shard.to(kwargs["device"]) for shard in self.get_shards()],
+        self.packed_shape,
+        self.bit_size,
+        self.pack_dim
+    )
+@implements([aten.to.dtype])
+def to(self, *args, **kwargs):
+    return self.__class__(
+        [shard.to(kwargs["dtype"]) for shard in self.get_shards()],
+        self.packed_shape,
+        self.bit_size,
+        self.pack_dim
+    )
+@implements([aten.t.default])
+def t(self,*args, **kwargs):
+    tensor = args[0]
+    new = tensor.__class__(
+        [shard.t() for shard in tensor.get_shards()],
+        tensor.packed_shape[::-1],
+        tensor.bit_size,
+        len(tensor.packed_shape) - tensor.pack_dim -1,
+    )
+    return return_and_correct_aliasing(func, args, kwargs, new)
+
+@implements([aten.detach.default])
+def detach(func, *args, **kwargs):
```

Comment:
btw these will be updated to:

`detach(func, types, args, kwargs)` after https://github.com/pytorch/ao/pull/588